bundles/specmate-integration-test/src/com/specmate/test/integration/ModelGenerationTestBase.java

```diff
@@ -0,0 +1,366 @@
+						|| (EmfRestTestUtil.matches(generated,
+								MATCHES_VAR_COND(firstLetterUpper(node.getVariable()),
+										firstLetterUpper(node.getCondition()), node.getType().getLiteral())))
+						|| (EmfRestTestUtil.matches(generated,
+								MATCHES_VAR_COND(firstLetterLower(node.getVariable()),
+										firstLetterLower(node.getCondition()), node.getType().getLiteral())))
+						|| (EmfRestTestUtil.matches(generated,
+								MATCHES_VAR_COND(node.getVariable(), firstLetterUpper(node.getCondition()),
+										node.getType().getLiteral())))
+						|| (EmfRestTestUtil.matches(generated,
+								MATCHES_VAR_COND(node.getVariable(), firstLetterLower(node.getCondition()),
+										node.getType().getLiteral())))
+						|| (EmfRestTestUtil.matches(generated, MATCHES_VAR_COND(node.getVariable(),
+								firstLetterLower(node.getCondition()), node.getType().getLiteral())));
+			}
+
+			Assert.assertTrue("Node with variable \"" + node.getVariable() + "\" and condition \"" + node.getCondition()
+					+ "\" not found. Nodes found: " + nodes, matched);
+		}
+
+		// Verify connections
+		for (CEGConnection conn : connExp) {
+			CEGNode expSource = (CEGNode) conn.getSource();
+			CEGNode expTarget = (CEGNode) conn.getTarget();
+			boolean matched = EmfRestTestUtil.matches(generated, obj -> {
+				if (!obj.getString(ECLASS).equals(CEGConnection.class.getSimpleName())) {
+					return false;
+				}
+				if (obj.getBoolean(RequirementsPackage.Literals.CEG_CONNECTION__NEGATE.getName()) != conn.isNegate()) {
+					return false;
+				}
+				JSONObject srcRef = obj.getJSONObject(BasePackage.Literals.IMODEL_CONNECTION__SOURCE.getName());
+				String srcId = srcRef.getString(URL);
+				srcId = srcId.substring(srcId.lastIndexOf("/") + 1);
+				JSONObject targetRef = obj.getJSONObject(BasePackage.Literals.IMODEL_CONNECTION__TARGET.getName());
+				String targetId = targetRef.getString(URL);
+				targetId = targetId.substring(targetId.lastIndexOf("/") + 1);
+				
+				return verifyConnections(expSource, srcId, expTarget, targetId, generated);
+			}
+
+			);
+
+			Assert.assertTrue("Connection from node (\"" + expSource.getVariable() + "\",\"" + expSource.getCondition()
+					+ "\")" + " to node (\"" + expTarget.getVariable() + "\",\"" + expTarget.getCondition()
+					+ "\")  not found. Negated: "+ conn.isNegate(), matched);
+		}
+
+	}
+
+	private String firstLetterUpper(String string) {
+		return string.substring(0, 1).toUpperCase() + string.substring(1);
+	}
+
+	private String firstLetterLower(String string) {
+		return string.substring(0, 1).toLowerCase() + string.substring(1);
+	}
+	
+	//Check all combination of Nodes(lower/upper case, with/without condition...)
+	private boolean verifyConnections(CEGNode expSource, String srcId,  CEGNode expTarget, String targetId, JSONArray generated) {
+		if (expSource.getCondition().equals("")) {
+			return (EmfRestTestUtil.matches(generated,

```

Comment:
wouldn't it be easier to compare everything converted to lowercase?
Anyway, this method is way to long, can you pull out methods and give them a reasonalbe name?